lib/generateClass.js

```diff
@@ -205,6 +213,65 @@ function writeFromJson(def, cw) {
   cw.ln('return inst;');
 }
 
+/**
+ * Generates a JSON serializer for the element
+ * @param {DataElement} def - The definition of the SHR element to generate a serializer for
+ * @param {CodeWriter} cw - The CodeWriter instance to use during generation
+ * @private
+ */
+function writeToJson(def, cw) {
+  // If the element is an Entry, put those fields on the JSON object first
+  if (def.isEntry) {
+    cw.ln(`const inst = this._entryInfo.toJSON();`);
+  } else if (def.identifier.name !== 'EntryType') {
+    const url = `http://standardhealthrecord.org/spec/${def.identifier.namespace.replace('.', '/')}/${def.identifier.name}`;
+    cw.ln(`const inst = { 'shr.base.EntryType': {'Value': '${url}'}};`);
+  } else {
+    cw.ln(`const inst = {};`);
+  }
+
+  if (def.value !== undefined) {
+    cw.bl('if (this.value !== undefined)', () => {
+      if (def.value instanceof ChoiceValue) {
+        // Choices get a special treatment
+        cw.ln(`inst['Value'] = typeof this.value.toJSON === 'function' ? this.value.toJSON() : this.value;`);
+        
+      } else if (def.value instanceof RefValue) {
+        // References get a special treatment too
+        cw.bl(`if (this.${toSymbol(def.value.identifier.name)} !== undefined)`, () => {
+          if (def.value.card.max > 1 || def.value.card.isMaxUnbounded) {
```

Comment:
Oops, totally missed that when I was looking at the `Cardinality` model. Thanks! 

Also, I'm still learning the difference between getters and functions, so that's good knowledge.

Reply:
I'm not sure there is a hard and fast rule about when something should be a getter vs when it should be a function, but I still feel like I probably made the wrong choice in a few spots in shr-models, so don't learn it from me!